### PR TITLE
gateway-api: Filter routes based on Section Name and port

### PR DESCRIPTION
**Pull request by sayboras**
### Description

Please refer to individual commit for more details.

### CODE REVIEW

File: operator/pkg/model/translation/gateway-api/translator.go [hunk 1]
```diff
@@ -90,7 +90,7 @@ func getService(resource *model.FullyQualifiedResource, allPorts []uint32) *core
 
 	return &corev1.Service{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      ciliumGatewayPrefix + resource.Name,
+			Name:      shorten(ciliumGatewayPrefix + resource.Name),
```
Comment (aanm) [inline operator/pkg/model/translation/gateway-api/translator.go:94]: We need to think about name collisions before shortening this name. Discussed offline, we could some hash to help out in here.

**Review by aanm — CHANGES_REQUESTED** [review]

File: operator/pkg/model/translation/gateway-api/translator.go (hunk 1 above)
Comment (sayboras) [reply operator/pkg/model/translation/gateway-api/translator.go:94]: thanks :100:

**Review by aanm — APPROVED** [review]

**Review by mhofstetter — APPROVED** [review]

**Comment by sayboras** [comment]
Reviews from required teams are in, merging this in.

**Pull request merged**